axum/src/extract/extractor_middleware.rs

```diff
@@ -2,324 +2,18 @@
 //!
 //! See [`extractor_middleware`] for more details.
 
-use super::{FromRequest, RequestParts};
-use crate::{
-    body::{Bytes, HttpBody},
-    response::{IntoResponse, Response},
-    BoxError,
-};
-use futures_util::{future::BoxFuture, ready};
-use http::Request;
-use pin_project_lite::pin_project;
-use std::{
-    fmt,
-    future::Future,
-    marker::PhantomData,
-    pin::Pin,
-    task::{Context, Poll},
+use crate::middleware::from_extractor;
+
+pub use crate::middleware::{
+    future::FromExtractorResponseFuture as ResponseFuture, FromExtractor as ExtractorMiddleware,
+    FromExtractorLayer as ExtractorMiddlewareLayer,
 };
-use tower_layer::Layer;
-use tower_service::Service;
 
 /// Convert an extractor into a middleware.
 ///
-/// If the extractor succeeds the value will be discarded and the inner service
-/// will be called. If the extractor fails the rejection will be returned and
-/// the inner service will _not_ be called.
-///
-/// This can be used to perform validation of requests if the validation doesn't
-/// produce any useful output, and run the extractor for several handlers
-/// without repeating it in the function signature.
-///
-/// Note that if the extractor consumes the request body, as `String` or
-/// [`Bytes`] does, an empty body will be left in its place. Thus wont be
-/// accessible to subsequent extractors or handlers.
-///
-/// # Example
-///
-/// ```rust
-/// use axum::{
-///     extract::{extractor_middleware, FromRequest, RequestParts},
-///     routing::{get, post},
-///     Router,
-/// };
-/// use http::StatusCode;
-/// use async_trait::async_trait;
-///
-/// // An extractor that performs authorization.
-/// struct RequireAuth;
-///
-/// #[async_trait]
-/// impl<B> FromRequest<B> for RequireAuth
-/// where
-///     B: Send,
-/// {
-///     type Rejection = StatusCode;
+/// Deprecated, please use [`crate::middleware::from_extractor`] instead
```

Comment:
This note is shown in the generated docs as well so we can probably just remove it. Its a bit redundant now.

<img width="646" alt="image" src="https://user-images.githubusercontent.com/718941/164973702-d275e26e-90b9-4b4e-8a09-508b2f9ed96d.png">